utils/fhir-server-export.py

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env python3
+"""Util to download all resources from the given FHIR server"""
+import requests, sys, time
+
+
+def download_file(url, filename=None):
+    """Download given large file via streaming"""
+    # https://stackoverflow.com/a/16696317
+    if not filename:
+        filename = url.split("/")[-1]
+    with requests.get(url, stream=True) as r:
+        r.raise_for_status()
+        with open(filename, "wb") as f:
+            for chunk in r.iter_content(chunk_size=8192):
+                # If you have chunk encoded response uncomment if
+                # and set chunk_size parameter to None.
+                # if chunk:
+                f.write(chunk)
+    return filename
+
+
+def fixup_url(url, base_url):
+    """
+    Replace FHIR base URL in given FHIR API call with different base_url
+    Helpful when a FHIR server is configured with a server_name that does not match its public one
+    """
+    if not url.startswith(base_url):
+        second_last_path, last_path = url.split("/")[-2:]
+        # FHIR operations
+        if last_path.startswith("$"):
+            return f"{base_url}/{last_path}"
```

Comment:
i don't know the spec as well as you, but is this a safe assumption?  do operations never have a named resource in the request, i.e. "http://base/fhir/Patient/$process"

Reply:
it's not generally a safe assumption. however, I haven't been able to figure out a consistent way to get the FHIR base URL, so I've been relying on some loose heuristics, specific to bulk export.

would love to hear any ideas, but will update the docstring to make it more obvious this process isn't perfect